primefaces/src/main/java/org/primefaces/model/filter/ComparableFilterConstraint.java

```diff
@@ -23,17 +23,61 @@
  */
 package org.primefaces.model.filter;
 
+import java.math.BigDecimal;
+import java.text.DecimalFormat;
+import java.text.NumberFormat;
+import java.text.ParseException;
 import javax.faces.context.FacesContext;
 import java.util.Locale;
+import java.util.function.BiPredicate;
 
-public abstract class ComparableFilterConstraint extends StringFilterConstraint {
+public abstract class ComparableFilterConstraint implements FilterConstraint {
 
     @Override
     public boolean isMatching(FacesContext ctxt, Object value, Object filter, Locale locale) {
+        if (value == null || filter == null) {
+            return false;
+        }
         if (!(value instanceof Comparable) || !(filter instanceof Comparable)) {
```

Comment:
We don't really need this check as the fallback is converting both sides to string